realm/realm-library/src/androidTestObjectServer/kotlin/io/realm/mongodb/sync/SessionTests.kt

```diff
@@ -392,21 +428,23 @@ class SessionTests {
 
     // Check that a Client Reset is correctly reported.
     @Test
-    fun errorHandler_manuallyRecoverClientResetReported() = looperThread.runBlocking {
+    fun errorHandler_manuallyRecoverUnsyncedChangesStrategy_errorHandled() = looperThread.runBlocking {
         val config = configFactory.createSyncConfigurationBuilder(user)
             .testSchema(SyncStringOnly::class.java)
-            .syncClientResetStrategy { session: SyncSession, error: ClientResetRequiredError ->
-                val filePathFromError = error.originalFile.absolutePath
-                val filePathFromConfig = session.configuration.path
-                assertEquals(filePathFromError, filePathFromConfig)
-                assertFalse(error.backupFile.exists())
-                assertTrue(error.originalFile.exists())
-                // Note, this error message is just the one created by ObjectStore for testing
-                // The server will send a different message. This just ensures that we don't
-                // accidentially modify or remove the message.
-                assertEquals("Simulate Client Reset", error.message)
-                looperThread.testComplete()
-            }
+            .syncClientResetStrategy(object: ManuallyRecoverUnsyncedChangesStrategy {
```

Comment:
The contract hasn't changed, but we can no longer instantiate the interface as a SAM because it conflicts with other interfaces. Shall we consider this a breaking change @cmelchior ?

Reply:
My bad, I misread the IDE exception. It is not a needed change.